packages/client/src/links/httpSubscriptionLink.ts

```diff
@@ -37,7 +36,17 @@ type HTTPSubscriptionLinkOptions<TRoot extends AnyClientTypes> = {
   /**
    * EventSource options or a callback that returns them
    */
-  eventSourceOptions?: CallbackOrValue<EventSourceInit>;
+  eventSourceOptions?: () => EventSourceInit | Promise<EventSourceInit>;
+  /**
+   * For a given error, should we reinitialize the underlying EventSource?
+   *
+   * This is useful where a long running subscription might be interrupted by a recoverable network error,
+   * but the existing authorization header might have expired in the mean-time
+   */
+  shouldReinitialize?: (
```

Comment:
naming is hard, just thinking of alternatives

```suggestion
  shouldRecreate?: (
```

```suggestion
  shouldRecreateOnError?: (
```

Reply:
Maybe `shouldRecreateOn4xx` even if we want it to be specific

Currently naming suggests it's generic but it's very specific

Reply:
`shouldRecreateOnError` I definitely prefer, think we can commit to that